extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/impl/ReadKafkaConnectP.java

```diff
@@ -55,9 +64,11 @@ public class ReadKafkaConnectP extends AbstractProcessor implements DynamicMetri
     private Traverser<?> traverser;
     private final LocalKafkaConnectStatsImpl localKafkaConnectStats = new LocalKafkaConnectStatsImpl();
 
-    public ReadKafkaConnectP(ConnectorWrapper connectorWrapper, EventTimePolicy<? super SourceRecord> eventTimePolicy) {
+    public ReadKafkaConnectP(ConnectorWrapper connectorWrapper, EventTimePolicy<? super SourceRecord> eventTimePolicy,
+                             @Nonnull FunctionEx<SourceRecord, T> projectionFn) {
```

Comment:
Maybe all could have the `@Nonnull/@Nullable` annotation.

More importantly, we should check for null here in the constructor. We sould get the NPE in `init` or some other method and it's hard to track down.

Reply:
Fixed https://github.com/hazelcast/hazelcast/pull/24203/commits/6a75fa0d9344412d420c3a4628cb206aec13dc30